### PR TITLE
nixos/dbus: Clean up

### PR DESCRIPTION
**Pull request by jtojnar**
###### Description of changes

cc @NixOS/freedesktop

###### Things done

- [x] Run GNOME VM with this change. 
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/system/dbus.nix [hunk 1]
```diff
@@ -102,10 +82,12 @@ in
...
     users.groups.messagebus.gid = config.ids.gids.messagebus;
 
-    systemd.packages = [ pkgs.dbus.daemon ];
+    systemd.packages = [
+      pkgs.dbus
+    ];
 
     security.wrappers.dbus-daemon-launch-helper = {
-      source = "${pkgs.dbus.daemon}/libexec/dbus-daemon-launch-helper";
+      source = "${lib.getBin pkgs.dbus}/libexec/dbus-daemon-launch-helper";
```
Comment (Atemu) [inline nixos/modules/services/system/dbus.nix:90]: Why is `getBin` necessary?

File: nixos/modules/services/system/dbus.nix [hunk 2]
```diff
@@ -65,31 +63,13 @@ in
         '';
         default = "disabled";
       };
-
-      socketActivated = mkOption {
```
Comment (Atemu) [inline nixos/modules/services/system/dbus.nix:69]: Shouldn't this be declared as a removed module?

File: nixos/modules/services/system/dbus.nix [hunk 3]
```diff
@@ -102,10 +82,12 @@ in
 
     users.groups.messagebus.gid = config.ids.gids.messagebus;
 
-    systemd.packages = [ pkgs.dbus.daemon ];
+    systemd.packages = [
+      pkgs.dbus
+    ];
```
Comment (Atemu) [inline nixos/modules/services/system/dbus.nix:87]: Why do we need the regular package rather than daemon here?

**Review by Atemu — COMMENTED** [review]
A few questions and nits, everything else LGTM

File: nixos/modules/services/system/dbus.nix (hunk 2 above)
Comment (jtojnar) [reply nixos/modules/services/system/dbus.nix:69]: It was declared as removed for the past two years. This pr removes that marking.

File: nixos/modules/services/system/dbus.nix (hunk 3 above)
Comment (jtojnar) [reply nixos/modules/services/system/dbus.nix:87]: See the commit message, there is no daemon package.

File: nixos/modules/services/system/dbus.nix (hunk 1 above)
Comment (jtojnar) [reply nixos/modules/services/system/dbus.nix]: It is not.

File: nixos/modules/services/system/dbus.nix (hunk 2 above)
Comment (Atemu) [reply nixos/modules/services/system/dbus.nix:69]: IIRC we have some specific list or option for removed options; this declaration was home-brewed.

**Review by Atemu — APPROVED** [review]

File: nixos/modules/services/system/dbus.nix (hunk 2 above)
Comment (jtojnar) [reply nixos/modules/services/system/dbus.nix:69]: Yes, that was used in the past: https://github.com/NixOS/nixpkgs/commit/f292a27f442d10de4827800d064a3a8c64d05cee
Then it was switched to the custom one: https://github.com/NixOS/nixpkgs/commit/94819fdb5f0b624fd4e8233a064c0eaa1479c170

Either way, the removal has been notified for a sufficient time now.

**Review by m-bdf — APPROVED** [review]
LGTM, though I'd suggest grouping the options as follows:
- `environment.systemPackages` with `security.wrappers.*` (these add stuff to PATH)
- `environment.etc.*` with `environment.pathsToLink` (these set up /etc and /share)
- `users.*`
- `systemd.packages` with the other `systemd.*` options
- `services.dbus.*`

**Review by xaverdh — APPROVED** [review]

**Comment by jtojnar** [comment]
@m-bdf I though about re-ordering but I am not aware of any conventions. Other than maybe having self-config options first. And since https://github.com/NixOS/nixpkgs/pull/122547 will re-order it anyway, I decided against it.

**Review by fabianhjr — APPROVED** [review]
LGTM, built without issues.

File: nixos/modules/services/system/dbus.nix [hunk 4]
```diff
@@ -16,11 +14,11 @@ let
     serviceDirectories = cfg.packages;
   };
 
+  inherit (lib) mkOption types;
```
Comment (SuperSandro2000) [inline nixos/modules/services/system/dbus.nix:17]: IMO not a fan of maintaining such inherits

**Comment by SuperSandro2000** [comment]
#202124

File: nixos/modules/services/system/dbus.nix (hunk 4 above)
Comment (Atemu) [reply nixos/modules/services/system/dbus.nix:17]: I think it's better than a global `with` or dotting `lib.` everywhere. It's a bit odd but I think it's the best option out of the bunch.

Comment (peterhoeg) [reply nixos/modules/services/system/dbus.nix:17]: I have had this same conversation with @SuperSandro2000 and I"m with @Atemu on this.

**Pull request merged**